File: examples/DesignPatterns/FactoryMethod/typescript/src/models/Action.ts
import { ActionType } from './ActionType';

export class Action<T> {
  private _type: ActionType;
  private _payload: T;

  constructor(type: ActionType, payload: T) {
    this._type = type;
    this._payload = payload;
  }

  public get type(): ActionType {
    return this._type;
  }

  public set type(type: ActionType) {
    this._type = type;
  }

  public get payload(): T {
    return this._payload;
  }
  public set payload(payload: T) {
    this._payload = payload;
  }
}

File: examples/DesignPatterns/FactoryMethod/src/main/java/models/Action.java
package models;

import enums.ActionType;

public class Action<T> {
    private ActionType type;
    private T payload;

    public Action(ActionType type, T payload) {
        this.type = type;
        this.payload = payload;
    }

    public ActionType getType() {
        return type;
    }

    public void setType(ActionType type) {
        this.type = type;
    }

    public T getPayload() {
        return payload;
    }

    public void setPayload(T payload) {
        this.payload = payload;
    }
}
